ddtrace/settings/peer_service.py

```diff
@@ -3,16 +3,27 @@
 from ddtrace.ext import SpanKind
 from ddtrace.internal.schema import SCHEMA_VERSION
 from ddtrace.internal.utils.formats import asbool
+from ddtrace.internal.utils.formats import parse_tags_str
 
 
 class PeerServiceConfig(object):
+    remap_tag_name = "_dd.peer.service.remapped_from"
     source_tag_name = "_dd.peer.service.source"
     tag_name = "peer.service"
     enabled_span_kinds = {SpanKind.CLIENT, SpanKind.PRODUCER}
     prioritized_data_sources = ["messaging.kafka.bootstrap.servers", "db.name", "rpc.service", "out.host"]
 
+    def __init__(self, enabled=None):
+        self._enabled = enabled
+
     @property
     def enabled(self):
-        env_enabled = asbool(os.getenv("DD_TRACE_PEER_SERVICE_DEFAULTS_ENABLED", default=False))
+        if self._enabled is None:
+            env_enabled = asbool(os.getenv("DD_TRACE_PEER_SERVICE_DEFAULTS_ENABLED", default=False))
+            self._enabled = SCHEMA_VERSION == "v1" or (SCHEMA_VERSION == "v0" and env_enabled)
+
+        return self._enabled
 
-        return SCHEMA_VERSION == "v1" or (SCHEMA_VERSION == "v0" and env_enabled)
+    @property
+    def peer_service_mapping(self):
+        return parse_tags_str(os.getenv("DD_TRACE_PEER_SERVICE_MAPPING", default=""))
```

Comment:
Does it make sense to have the same pattern as `enabled` here to avoid accessing the environment every time we want the mapping?

Reply:
Updating